### PR TITLE
lmp-auto-hostname: improving user guide

### PR DESCRIPTION
**Pull request by munoz0raul**
Signed-off-by: Raul Muñoz <raul@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 4f7d02dcf6979d986c7becffc8ad3813a82d87b8 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1109/docs/artifacts/html/index.html

File: source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+The latest **Target** named ``platform-devel`` should be the CI job you just created.
+
+When FoundriesFactory CI finishes all jobs, if your device is already registered, 
+wait until the Over-the-Air update finishes, otherwise download and flash the image.
+
+Testing Auto Hostname
+^^^^^^^^^^^^^^^^^^^^^
+
+Log in to the device via SSH and it will be possible to see right after ``fio@`` the new hostname.
```
Comment (vanmaegima) [inline source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst:134]:
```suggestion
Log in to the device via SSH and check the new hostname right after ``fio@``.
```

**Comment by doanac** [comment]
Docs for f157f370acfa0f29a6365a4bc5ae55ca2cc8edc6 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1111/docs/artifacts/html/index.html

**Review by vanmaegima — APPROVED** [review]
LGTM

File: source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      
+      .. prompt:: text
+      
+           raspberrypi3-64-b827ebca7875
+
+Summary
+^^^^^^^
+
+This user guide gives you an example of how to enable the LmP Auto Hostname. 
+It helps when each device needs a custom hostname.
```
Comment (angolini) [inline source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst:168]: I would drop the summary for the UG pages.

File: source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``lmp-auto-hostname`` recipe can be configured through variables.
+
+.. confval:: LMP_HOSTNAME_MODE=<option>
+    :default: ``serial``
+
+    .. option:: serial
+
+       appends the serial number of the device.
```
Comment (angolini) [inline source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst:65]: I'm not sure I understand which serial number is it. Would it be some FF serial number? Or the SoC serial number?

**Review by angolini — APPROVED** [review]
I think it really improves the UG, I like how you change the filenames being included.

I would think about keeping the esthetic for the pages from UG and from Tutorial different. I love that we have know a Tutorial section because they are really special, and I concern bringing the same elements to the other pages could fade Tutorials somehow, what do you think?

File: source/index.rst [hunk 4]
```diff
@@ -52,8 +52,10 @@ OE/Yocto Project, the Linux microPlatform™ and Docker®.
    :caption: User Guide
    :name: sec-user-guide
 
-   user-guide/*/*
-
+   user-guide/configure-lmp/index
```
[the remark this answers is not in the document]
Comment (munoz0raul) [reply source/index.rst:55]: Let's get this PR approved and create a new one fixing the structure.

**Comment by doanac** [comment]
Docs for 1a8a1443f7aff183cce85ef38e0f71df4e9d3b16 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/1117/docs/artifacts/html/index.html

File: source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst (hunk 3 above)
Comment (munoz0raul) [reply source/user-guide/lmp-auto-hostname/lmp-auto-hostname.rst:55]: https://github.com/foundriesio/meta-lmp/blob/master/meta-lmp-base/recipes-support/lmp-auto-hostname/lmp-auto-hostname/lmp-update-hostname.sh#L23

I think at this point if the customer wants more detail, he needs to take a look at the reference manual or read the recipe.
Don't want to extend too much the UG.

**Review by angolini — APPROVED** [review]

**Pull request merged**